src/BSON/Decimal128.c

```diff
@@ -46,34 +46,39 @@ PHONGO_API zend_class_entry *php_phongo_decimal128_ce;
 
 zend_object_handlers php_phongo_handler_decimal128;
 
-/* Initialize the object from a string and return whether it was successful. */
-static bool php_phongo_decimal128_init(php_phongo_decimal128_t *intern, const char *value)
+/* Initialize the object and return whether it was successful. An exception will
+ * be thrown on error. */
+static bool php_phongo_decimal128_init(php_phongo_decimal128_t *intern, const char *value TSRMLS_DC)
 {
-	if (bson_decimal128_from_string(value, &intern->decimal)) {
-		intern->initialized = true;
-
-		return true;
+	if (!bson_decimal128_from_string(value, &intern->decimal)) {
+		phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Error parsing decimal string: %s", value);
```

Comment:
This is also a message that might benefit from adding the class name.

On a side note, this differs from HHVM's exception message (see: [here](https://github.com/mongodb/mongo-hhvm-driver/blob/73bc3daa7691184a6e091585ad42c01cb965fe41/src/MongoDB/BSON/Decimal128.cpp#L39)). We must have missed the different when originally reviewing [PHPC-619](https://jira.mongodb.org/browse/PHPC-619)'s PR.

Let's take this opportunity to settle on a consistent format.